src/components/PupilViews/PupilQuiz/PupilQuiz.view.tsx

```diff
@@ -1,28 +1,93 @@
-import { OakFlex } from "@oak-academy/oak-components";
+import {
+  OakBackLink,
+  OakFlex,
+  OakLessonBottomNav,
+  OakLessonLayout,
+  OakLessonTopNav,
+  OakPrimaryButton,
+} from "@oak-academy/oak-components";
 
 import {
   QuestionsArray,
   QuizEngineProvider,
+  useQuizEngineContext,
 } from "@/components/PupilComponents/QuizEngineProvider";
 import { QuizRenderer } from "@/components/PupilComponents/QuizRenderer";
+import { useLessonEngineContext } from "@/components/PupilComponents/LessonEngineProvider";
 
 type PupilViewsQuizProps = {
   questionsArray: QuestionsArray;
 };
 
-export const PupilViewsQuiz = ({ questionsArray }: PupilViewsQuizProps) => {
+type CurrentSection = "starter-quiz" | "exit-quiz";
+
+const QuizInner = () => {
+  const { currentSection, updateCurrentSection } = useLessonEngineContext();
+  const quizEngineContext = useQuizEngineContext();
+
+  const { currentQuestionIndex, questionState, handleNextQuestion } =
+    quizEngineContext;
+
+  const isFeedbackMode =
+    questionState[currentQuestionIndex]?.mode === "feedback";
+
   return (
-    <QuizEngineProvider questionsArray={questionsArray}>
-      <OakFlex
-        $width={"100vw"}
-        $height={"100vh"}
-        $background={"bg-decorative1–main"}
-        $flexDirection={"column"}
-        $alignItems={"center"}
-        $pt="inner-padding-xl"
-      >
+    <OakLessonLayout
+      bottomNavSlot={
+        <OakLessonBottomNav>
+          {!isFeedbackMode && (
+            <OakFlex $pt="inner-padding-l">
+              <OakPrimaryButton
+                form={"a-form"}
+                disabled={questionState[currentQuestionIndex]?.mode === "init"}
+                type="submit"
+              >
+                Submit
+              </OakPrimaryButton>
+            </OakFlex>
+          )}
+          {isFeedbackMode && (
+            <OakFlex $pt="inner-padding-l">
+              <OakPrimaryButton onClick={handleNextQuestion}>
+                Next Question
+              </OakPrimaryButton>
+            </OakFlex>
+          )}
+        </OakLessonBottomNav>
+      }
+      lessonSectionName={currentSection as CurrentSection}
```

Comment:
try to avoid type assertion as it really is no protection at all. In this case you could write a simple conditional or write a custom type guard for CurrentSection (which should be renamed)

Reply:
@kimon-satan  youre taking this one right?